packages/api-sdk/src/medical/client/metriport.ts

```diff
@@ -211,14 +213,19 @@ export class MetriportMedicalApi {
   }
 
   /**
-   * Searches for a patient.
+   * Searches for a patient previously created at Metriport, based on demographics.
    *
    * @return The patient if found.
    */
-  async searchPatient(data: PatientSearch): Promise<PatientDTO> {
-    const resp = await this.api.post(`${PATIENT_URL}/search`, data);
-    if (!resp.data) throw new Error(NO_DATA_MESSAGE);
-    return resp.data as PatientDTO;
+  async matchPatient(data: Demographics): Promise<PatientDTO | undefined> {
+    try {
+      const resp = await this.api.post(`${PATIENT_URL}/match`, data);
+      return resp.data as PatientDTO;
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
```

Comment:
the general pattern in the SDKs is to use something like

`if (!resp.data) throw new Error `
Not saying this is wrong, but unless there is a good reason lets stick to that pattern?

Reply:
The idea was to catch the 404 in particular and return undefined from the SDK to make it a bit more usable I guess (it was a request from @leite08 in one of the resolved conversations).

But, we don't have a consistent approach for handling 404s across any other endpoint (i.e. `async getPatient(id: string): Promise<PatientDTO> {` does not catch a 404), so it's introducing a bit of a new approach for this one in particular.

@leite08 can you maybe describe why returning undefined in the 404 case for this one in particular makes sense?

Reply:
Yeah, those are diff situations and it seems a good idea to check if data is empty.

The usual pattern @jonahkaye referred to doesn't seem to apply here, as its specifically directed to the case where we're always returning something (no `| undefined` on the response type).

I'd argue we might as well return undefined it no `resp.data`, kinda a defensive programming approach - let's say we setup Axios to not throw on 404, then the SDK still behaves properly.

> can you maybe describe why returning undefined in the 404 case for this one in particular makes sense?

As a developer using the SDK, I ask to match demographics; if it does, return me the patient, otherwise return undefined. I want to handle errors if an error happened, 404 is not an error on this use case, right?

Reply:
@leite08  to be clear: this endpoint *will* throw a 404 if no patient is found. Maybe that's the confusion?

Reply:
> this endpoint will throw a 404 if no patient is found. Maybe that's the confusion?

I know it _returns_ a 404. Axios is the one that throws by default on 4xx/5xx.

I just don't see a confusion.

If I'm calling an API and it returns 404, I know it didn't find what I was looking for - better this than making me parse the payload.

But if I'm using a Typescript SDK, I expect that not found returns `undefined`, not `Error`.

Reply:
We could argue that the SDK as a whole should prevent throwing `Error` on 404, but that seems a larger convo - this case it clearly can return `undefined`.

Reply:
The endpoint does `throw` the 404 as I edited the route the code as follows:
`
    if (patient) {
      return res.status(status.OK).json(dtoFromModel(patient));
    }
    throw new NotFoundError("Cannot find patient");
`
Maybe this is the confusion?

Reply:
@leite08 any thoughts ☝️ ?

Reply:
I'm referring from the user's perspective:
- Calling the API I can get a `404` ("the API returns 404")
- Using the SDK, it returns `Patient` or `undefined`

That sounds reasonable and expected, so there's no error handling from my side as a client.

Thoughts?

Reply:
Offline discussion outcome:
1. We will leave the code with the 404 try/catch as we agreed undefined is a better SDK behavior than axios throwing the 404.
2. We will not address other endpoints that throw a 404 for now.